services/authentication.go

```diff
@@ -0,0 +1,79 @@
+package services
+
+import (
+	"context"
+	"encoding/base64"
+	"net/http"
+	"strings"
+
+	"github.com/kacperf531/sockchat/api"
+)
+
+type SockchatAuthService struct {
+	UserProfiles api.SockchatProfileStore
+}
+
+type authWrapper struct {
+	Username string
+	Password string
+}
+
+func (s *SockchatAuthService) AuthenticateFromBasicToken(ctx context.Context, token string) (bool, error) {
+	if !isBasicToken(token) {
+		return false, api.ErrBasicTokenRequired
+	}
+	auth, err := decodeToken(token)
+	if err != nil {
+		return false, err
+	}
+	return s.UserProfiles.IsAuthValid(ctx, auth.Username, auth.Password), nil
+}
+
+func isBasicToken(token string) bool {
+	return strings.HasPrefix(token, "Basic ")
+}
+
+func decodeToken(token string) (*authWrapper, error) {
+	encoded := strings.TrimPrefix(token, "Basic ")
+	data, err := base64.StdEncoding.DecodeString(encoded)
+	if err != nil {
+		return nil, api.ErrCouldNotDecodeToken
+	}
+	if !strings.Contains(string(data), ":") {
+		return nil, api.ErrBasicTokenRequired
+	}
+	nick := strings.Split(string(data), ":")[0]
+	pwHash := strings.Split(string(data), ":")[1]
+	return &authWrapper{nick, pwHash}, nil
```

Comment:
```suggestion
	credentials := strings.SplitN(string(data), ":", 2)
	if len(credentials) != 2 {
		return nil, api.ErrBasicTokenRequired
	}
	return &authWrapper{credentials[0], credentials[1]}, nil
```
It could be simplified with the [strings.SplitN](https://pkg.go.dev/strings#SplitN)